codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -511,4 +513,20 @@ public static CharSequence getMimeType(CharSequence contentTypeValue) {
             return contentTypeValue.length() > 0 ? contentTypeValue : null;
         }
     }
+
+    /**
+     * Formats an address so it can be used for computing an HTTP component such as an URL or a Host header
+     * @param addr the address
+     * @return the formatted String
+     */
+    public static String formatForHttp(InetSocketAddress addr) {
```

Comment:
I guess a user of this method may think the return value will contain the port number. How about adding a boolean parameter `withPort`, so that this method adds `:<port>` when it's `true`?

Reply:
What about trying to find a better name instead? Maybe `formatHostnameForHttp`?
I'm not fond of expanding an API beyond what's being used internally so it can serve for whatever other purpose we're not aware of.
WDYT?

Reply:
That's also fine.

Reply:
Done